src/decoder/mod.rs

```diff
@@ -303,6 +318,19 @@ fn predict_f32(input: &mut [u8], output: &mut [u8], samples: usize) {
     }
 }
 
+fn predict_f16(input: &mut [u8], output: &mut [u8], samples: usize) {
+    for i in samples..input.len() {
+        input[i] = input[i].wrapping_add(input[i - samples]);
+    }
+
+    for (i, chunk) in output.chunks_mut(2).enumerate() {
+        chunk.copy_from_slice(&u16::to_ne_bytes(u16::from_be_bytes([
+            input[i],
+            input[input.len() / 4 + i],
```

Comment:
Shouldn't this be `input.len() / 2 + i`?

Reply:
good catch copy-paste error from fp32. Unfortunately I don't have an fp16 image with prediction to test with. Wonder if I can generate one somehow. 

I'm also not entirely sure the prediction for any of the floating points make sense. Should 

```
 input[i] = input[i].wrapping_add(input[i - samples]);
```
really be operating on each byte? Shouldn't it operate on the words?